src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -89,40 +91,80 @@ public void update()
             {
                 if (i == index)
                 {
-                    if (buildingEntry.getValue() instanceof AbstractBuildingWorker
-                          && !list.contains(buildingEntry.getValue())
-                          && ((AbstractBuildingWorker) buildingEntry.getValue()).needsAnything())
+                    if(buildingEntry.getValue() instanceof AbstractBuildingWorker
+                            && !list.contains(buildingEntry.getValue())
+                            && ((AbstractBuildingWorker) buildingEntry.getValue()).needsAnything())
                     {
                         checkInWareHouse((AbstractBuildingWorker) buildingEntry.getValue(), true);
                     }
+                    else if(buildingEntry.getValue() instanceof BuildingHome && ((BuildingHome) buildingEntry.getValue()).isFoodNeeded())
+                    {
+                        checkInWareHouseForFood((BuildingHome) buildingEntry.getValue(), true);
+                    }
                     this.index++;
                 }
                 i++;
             }
         }
     }
 
+    public boolean checkInWareHouseForFood(final BuildingHome buildingEntry, final boolean addToList)
+    {
+        if (buildingEntry.isFoodNeeded())
+        {
+            if (isInHut(itemStack -> !InventoryUtils.isItemStackEmpty(itemStack) && itemStack.getItem() instanceof ItemFood))
+            {
+                if (addToList)
+                {
+                    buildingEntry.setOnGoingDelivery(true);
+                    list.add(buildingEntry);
+                }
+                return true;
+            }
+
+            if (list.contains(buildingEntry))
+            {
+                list.remove(buildingEntry);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "remove()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)